tests/integration/test_lineage.py

```diff
@@ -103,48 +116,234 @@ def test_get_column_lineage(config, ctx):
+    assert downstreams == expected_downstreams, f"check: {downstreams}"
+
+    # check that the view schema and name are correct
+    assert column_lineage[0][0].dataset.schema == "hotel_schema"
+    assert column_lineage[0][0].dataset.name == "guests"
+
+    # now check that each of these columns has the correct upstream (source) columns
+    upstream_field_names = [
+        ["name"],  # name of hotel
+        ["type"],  # type of room
+        ["price"],  # price of room
+        ["free"],  # status of room
+        # columms used in subselect to get num_reservations column
+        ["hno", "hno", "type", "type"],
+        ["name", "price"],  # cols from view total_room_price to get rn column
+    ]
+    # note that we sort the upstreams because the order is not guaranteed by sqlglot
+    upstreams = [sorted([source.name for source in x[1]])
+                 for x in column_lineage]
+
+    assert upstreams == upstream_field_names, f"found: {upstreams}"
+
+
+# test for cross-schema where the downstream is on reservations_schema and the upstreams are on hotel_schema
+@pytest.mark.db
+def test_cross_schema(testdata: Path, ctx):
+    scheme = "hana"  # make this into dictionary
+    username = "system"
+    password = "HXEHana1"
+    at = "localhost:39041"
+    db = "HXE"
+
+    # create dict for config
+    config_dict = {
+        "username": "system",
+        "password": "HXEHana1",
+        "host_port": "localhost:39041",
+        "database": "HXE",
+        "include_view_lineage": "true",
+        "include_column_lineage": "true",
+        "schema_pattern": AllowDenyPattern(allow=["RESERVATIONS*"]),
+    }
+
+    test_schema_file = testdata / "test_cross_schema.sql"
+    test_schema = test_schema_file.read_text()
+
+    uri = make_sqlalchemy_uri(scheme, username, password, at, db)
+    engine = create_engine(uri)
+    conn = engine.connect()
+
+    schema_results = conn.execute(
+        "SELECT * FROM SCHEMAS WHERE SCHEMA_NAME = 'RESERVATIONS_SCHEMA'"
+    )
+    print(schema_results)
+
+    if schema_results.returns_rows:
+        conn.execute("DROP SCHEMA RESERVATIONS_SCHEMA CASCADE")
+
+    conn.execute("CREATE SCHEMA RESERVATIONS_SCHEMA")
+    conn.execute(test_schema)
+
+    assert any(
+        "RESERVATIONS_SCHEMA" in item[0] for item in schema_results
+    ), f"{schema_results}"
+
+    # source class
+    source = HanaSource.create(config_dict, ctx)
+    source.get_db_connection()
+    inspector = inspect(conn)
+
+    view_definitions = source.get_column_lineage_view_definitions(inspector)
+
+    # sql = """SELECT
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE,
+    # COUNT(R.FREE) AS FREE_RM_COUNT
+    # FROM
+    # HOTEL_SCHEMA.ROOM AS R
+    # LEFT JOIN
+    # HOTEL_SCHEMA.HOTEL AS H
+    # ON H.HNO = R.HNO
+    # GROUP BY
+    # H.NAME,
+    # R.TYPE,
+    # R.FREE;"""
+
+    for views in view_definitions:
+        # pass
+        assert views.name == "test_cross_schema"
+        assert views.schema == "reservations_schema"
```

Comment:
this is a bit awkward, since you are essentially asserting that there i only one view right?

This is a bit more clear and specific 
```suggestion
    assert len(view_definitions) == 1
    view = view_definitions[0]
    assert view.name == "test_cross_schema"
    assert view.schema == "reservations_schema"
```